### PR TITLE
[UFCFC-36] Feature - Add Seek Intervals [Android] [iOS]

### PR DESCRIPTION
**Pull request by David06A**
Adds changeable seek intervals into VOD & Live Video through props for Android and iOS

"As a user watching a VOD or Live Video and DVR is available and I press the rewind button the video should seek backwards the interval determined by the realm settings"

Branch is currently pointing at dice unity pr: https://github.com/DiceTechnology/dice-unity/pull/628

[View in Jira](https://dicetech.atlassian.net/browse/UFCFC-36) 
----

**Videos**
### Android Sim
https://github.com/DiceTechnology/dice-tv/assets/74801783/8ffe81f4-4f1e-4ec9-96a9-819776a452ca

### iOS sim
https://github.com/DiceTechnology/dice-tv/assets/74801783/c24cb58a-8d28-4929-b0dc-0381d31b7b61

### Web
https://github.com/DiceTechnology/dice-tv/assets/74801783/706c7f9b-2428-40a0-929b-34ef5160d357

----

**Tested On**

- FireTV Physical Device
- Android Sim
- iOS sim

### CODE REVIEW

File: android-exoplayer/src/main/java/com/brentvatne/exoplayer/ReactTVExoplayerView.java [hunk 1]
```diff
@@ -1525,6 +1528,22 @@ public void setPlayInBackground(boolean playInBackground) {
...
+        this.dvrSeekBackwardInterval = seekBackwardMs;
+    }
+
+    public long getDVRSeekForward() {
+        return this.dvrSeekForwardInterval;
+    }
+
+    public long getDVRSeekBackward() {
+        return this.dvrSeekBackwardInterval;
+    }
```
Comment (oskr-franco) [inline android-exoplayer/src/main/java/com/brentvatne/exoplayer/ReactTVExoplayerView.java]: where are we using this?

**Comment by yaroslavlvov** [comment]
please move and read these two props into **src** and NOT as a separate prop
dvrSeekForwardInterval
dvrSeekForwardInterval

File: android-exoplayer/src/main/java/com/brentvatne/exoplayer/ReactTVExoplayerView.java (hunk 1 above)
Comment (David06A) [reply android-exoplayer/src/main/java/com/brentvatne/exoplayer/ReactTVExoplayerView.java]: Outdated due to props moving to source

File: android-exoplayer/src/main/java/com/brentvatne/exoplayer/ReactTVExoplayerView.java [hunk 2]
```diff
@@ -481,8 +481,8 @@ private void doInitializePlayer(boolean force) {
                     getContext(),
                     adType,
                     MAX_LOAD_BUFFER_MS,
-                    exoDorisPlayerView.getFastForwardIncrementMs(),
-                    exoDorisPlayerView.getRewindIncrementMs(),
+                    (src.getDvrSeekForwardInterval() != 0L) ? src.getDvrSeekForwardInterval() : exoDorisPlayerView.getFastForwardIncrementMs(),
```
Comment (oskr-franco) [inline android-exoplayer/src/main/java/com/brentvatne/exoplayer/ReactTVExoplayerView.java]: perhaps we can move them into variables and we can avoid executing `getDvrSeekForwardInterval()` twice

Comment (David06A) [reply android-exoplayer/src/main/java/com/brentvatne/exoplayer/ReactTVExoplayerView.java]: Agreed, changed

**Review by lorran-xo — DISMISSED** [review]

File: android-exoplayer/src/main/java/com/brentvatne/entity/RNSource.java [hunk 3]
```diff
@@ -86,6 +90,15 @@ public RNSource(
         this.yoSsai = yoSsai;
         this.limitedSeekRange = limitedSeekRange;
         this.tracksPolicy = tracksPolicy;
+        this.dvrSeekForwardInterval = convertToMillis(dvrSeekForwardInterval);
```
Comment (oskr-franco) [inline android-exoplayer/src/main/java/com/brentvatne/entity/RNSource.java]: I believe there is a function for this: `TimeUnit.SECONDS.toMillis`

Comment (David06A) [reply android-exoplayer/src/main/java/com/brentvatne/entity/RNSource.java]: Nice one thank you, added

**Review by lorran-xo — DISMISSED** [review]

**Review by oskr-franco — DISMISSED** [review]

File: package.json [hunk 4]
```diff
@@ -20,8 +20,8 @@
         "eslint-plugin-react": "3.16.1"
     },
     "dependencies": {
-        "@dicetechnology/react-native-dice-video": "git+ssh://git@github.com:DiceTechnology/react-native-dice-video.git#6.43.6",
-        "@dicetechnology/dice-unity": "^2.26.3",
+        "@dicetechnology/react-native-dice-video": "git+ssh://git@github.com:DiceTechnology/react-native-dice-video.git#6.48.0",
+        "@dicetechnology/dice-unity": "https://github.com/DiceTechnology/dice-unity.git#8a16345",
```
Comment (aceew) [inline package.json]: Let's remember to revert this

**Review by aceew — DISMISSED** [review]

File: package.json (hunk 4 above)
Comment (guoen21) [reply package.json:24]: I think we need to bump the dice-unity or revert it, instead of use commit hash in a PR.

**Review by guoen21 — DISMISSED** [review]

**Comment by guoen21** [comment]
In this PR we also need to moidfy the [package.json](https://github.com/DiceTechnology/react-native-video/blob/f747854da359ff6d1d1143b997ac78c76f2eeb1f/package.json#L3) to bump the `rnv` version to `7.3.7` or `7.4.0`, the latest version is `7.3.6` .

**Review by aceew — DISMISSED** [review]

**Review by yaroslavlvov — DISMISSED** [review]

**Review by yaroslavlvov — DISMISSED** [review]

**Review by yaroslavlvov — APPROVED** [review]

**Pull request merged**